### PR TITLE
fix: Thanos Sidecar object storage config documentation

### PR DESCRIPTION
**Pull request by bvis**
## Description

There's a missing update in the `Documentation/thanos.md` file related to the values needed for configure the object storage. 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
- Fix object storage config documentation for Thanos sidecar.
```

### CODE REVIEW

File: Documentation/thanos.md [hunk 1]
```diff
@@ -73,8 +73,9 @@ spec:
   thanos:
     image: quay.io/thanos/thanos:v0.28.1
     objectStorageConfig:
-      key: thanos.yaml
-      name: thanos-objstore-config
+      existingSecret:
```
[the remark this answers is not in the document]
Comment (bvis) [reply Documentation/thanos.md:76]: This is how this works right now in Helm, in fact there's the same change explained in the upgrade notes, but not in the related docs here: https://github.com/prometheus-community/helm-charts/blob/123dad00b55924592746af876ee1101c969b2168/charts/kube-prometheus-stack/README.md?plain=1#L110

[the remark this answers is not in the document]
Comment (bvis) [reply Documentation/thanos.md:76]: I got your point.

You are right that this is about the CRD and not how's this defined in the Helm chart, but from a user perspective it's confusing.

It has changed at Helm level but not in the CRD and both share the same doc block.

I suggest to reject this PR.

[the remark this answers is not in the document]
Comment (simonpasquier) [reply Documentation/thanos.md:76]: Yes, historically this repository doesn't prescribe a particular deployment method so we don't want to carry Helm specific instructions.

**Comment by ArthurSens** [comment]
Are you ok with closing this PR @bvis? :)

**Pull request closed without merging**